<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>709</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>0</y>
      <width>211</width>
      <height>501</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="MazeTab">
     <attribute name="title">
      <string>Maze</string>
     </attribute>
     <widget class="QSpinBox" name="m_cols">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>40</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>25</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="m_rows">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>51</width>
        <height>31</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>25</number>
      </property>
     </widget>
     <widget class="QPushButton" name="m_SaveFile">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>390</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Save file</string>
      </property>
     </widget>
     <widget class="QPushButton" name="m_OpenFile">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>430</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Open file</string>
      </property>
     </widget>
     <widget class="QPushButton" name="m_Generate">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>350</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string> Generate</string>
      </property>
     </widget>
     <widget class="Line" name="line_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>80</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Rows</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>80</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Cols</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="CaveTab">
     <attribute name="title">
      <string>Cave</string>
     </attribute>
     <widget class="QSpinBox" name="c_life">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_dead">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
     <widget class="QPushButton" name="Step">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>290</y>
        <width>161</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Step</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_rows">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>90</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_cols">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>90</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_popul">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>170</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_life_2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>10</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>8</number>
      </property>
      <property name="value">
       <number>7</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_dead_2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>40</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>8</number>
      </property>
      <property name="value">
       <number>7</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="life_change">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>200</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="value">
       <number>47</number>
      </property>
     </widget>
     <widget class="QPushButton" name="Time">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>260</y>
        <width>81</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Timer</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="c_second">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>260</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="minimum">
       <number>10</number>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
      <property name="singleStep">
       <number>5</number>
      </property>
     </widget>
     <widget class="QLabel" name="crowtxt">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Rows</string>
      </property>
     </widget>
     <widget class="QLabel" name="ccolstxt">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>110</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Cols</string>
      </property>
     </widget>
     <widget class="QLabel" name="lidetxt">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>10</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>survival</string>
      </property>
     </widget>
     <widget class="QLabel" name="deadtxt">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>40</y>
        <width>41</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>born </string>
      </property>
     </widget>
     <widget class="QLabel" name="popultxt">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>170</y>
        <width>81</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Populations</string>
      </property>
     </widget>
     <widget class="QLabel" name="chancetxt">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>200</y>
        <width>81</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Life chance</string>
      </property>
     </widget>
     <widget class="QPushButton" name="c_Generate">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>350</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string> Generate</string>
      </property>
     </widget>
     <widget class="QPushButton" name="c_OpenFile">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>430</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Open file</string>
      </property>
     </widget>
     <widget class="QPushButton" name="c_SaveFile">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>390</y>
        <width>111</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Save file</string>
      </property>
     </widget>
     <widget class="Line" name="line_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="Line" name="line_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>181</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGraphicsView" name="graph">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
